src/hotspot/share/utilities/exceptions.cpp

```diff
@@ -351,17 +351,15 @@ Handle Exceptions::new_exception(JavaThread* thread, Symbol* name,
   if (message == NULL) {
     signature = vmSymbols::void_method_signature();
   } else {
-    // We want to allocate storage, but we can't do that if there's
-    // a pending exception, so we preserve any pending exception
-    // around the allocation.
-    // If we get an exception from the allocation, prefer that to
-    // the exception we are trying to build, or the pending exception.
-    // This is sort of like what PreserveExceptionMark does, except
-    // for the preferencing and the early returns.
-    Handle incoming_exception(thread, NULL);
+    // There should be no pending exception. The caller is responsible for not calling
+    // this with a pending exception.
+    Handle incoming_exception;
     if (thread->has_pending_exception()) {
       incoming_exception = Handle(thread, thread->pending_exception());
       thread->clear_pending_exception();
+      incoming_exception->print();
```

Comment:
We shouldn't be unconditionally printing in product mode. If we have an unexpected pending exception then that is a bug in the VM code. The end user won't have a clue what is being printed or why, nor what to do about it.

Reply:
Yes, I agree. I was going to add it under -Xlog:exceptions but if  you turn on -Xlog:exceptions, you can see where the pending exception comes from so that's not helpful. I removed it.